datahub-frontend/app/auth/AuthModule.java

```diff
@@ -89,7 +89,7 @@ protected void configure() {
                 // it to hex and slice the first 16 bytes, because AES key length must strictly
                 // have a specific length.
                 final String aesKeyBase = _configs.getString(PAC4J_AES_KEY_BASE_CONF);
-                final String aesKeyHash = DigestUtils.sha1Hex(aesKeyBase.getBytes(StandardCharsets.UTF_8));
+                final String aesKeyHash = DigestUtils.sha256Hex(aesKeyBase.getBytes(StandardCharsets.UTF_8));
                 final String aesEncryptionKey = aesKeyHash.substring(0, 16);
```

Comment:
Why only 16 bytes?

Reply:
AES key length is limited to 16 bytes, per comment above. NOTE: This is totally separate from the HMAC signing. This is for data encryption. I'm fairly certain it's unused since we don't set data encryption for the cookies we use, but I couldn't tell from the Scala code where precisely it gets used. All I could tell is that it doesn't have an impact on the HMAC signature algorithm (that's using HS256 and AES is not involved).

Best I could tell it encrypts the values inside the PlayWebContext, but the values are not encrypted in the JWT (I can put the JWT in JWT.io's debugger and see the values and grab them directly as exemplified in the test I modified). Why it does this I'm not sure: https://github.com/pac4j/play-pac4j/blob/master/shared/src/main/java/org/pac4j/play/store/PlayCookieSessionStore.java